client/post-editor/post-editor.jsx

```diff
@@ -650,14 +649,12 @@ export const PostEditor = React.createClass( {
 
 	onPublish: function() {
 		const edits = {
-			...this.props.edits,
-			status: 'publish'
+			status: 'publish',
+			...this.props.edits
 		};
 
-		// determine if this is a private publish
-		if ( utils.isPrivate( this.state.post ) ) {
-			edits.status = 'private';
-		} else if ( utils.isFutureDated( this.state.post ) ) {
+		// determine if this is a future publish
+		if ( utils.isFutureDated( this.state.post ) ) {
```

Comment:
After some tests, I'm leaving this as is for now. If we were to make it explicit we'll need to undo the `status` edit when the date is set to a previous date, which makes me think the most convenient way to handle this is on the `onPublish` or server side like now, edits should only contain user edits.